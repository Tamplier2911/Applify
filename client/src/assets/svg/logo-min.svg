<?xml version="1.0" standalone="no"?>
<!DOCTYPE svg PUBLIC "-//W3C//DTD SVG 20010904//EN"
 "http://www.w3.org/TR/2001/REC-SVG-20010904/DTD/svg10.dtd">
<svg version="1.0" xmlns="http://www.w3.org/2000/svg"
 width="500.000000pt" height="500.000000pt" viewBox="0 0 500.000000 500.000000"
 preserveAspectRatio="xMidYMid meet">
<metadata>
Created by potrace 1.15, written by Peter Selinger 2001-2017
</metadata>
<g transform="translate(0.000000,500.000000) scale(0.100000,-0.100000)"
 stroke="none">
<path d="M1074 4569 c-8 -13 -7 -1759 1 -1767 6 -6 104 26 118 38 4 4 8 365 7
803 l0 797 1620 0 1620 0 -3 -1612 -2 -1613 -802 -3 c-756 -2 -803 -3 -804
-20 0 -9 -3 -38 -5 -64 -4 -36 -2 -48 8 -48 7 0 290 0 628 0 338 -1 729 -1
868 0 l252 0 0 1744 c0 1581 -1 1744 -16 1750 -30 12 -3482 7 -3490 -5z"/>
<path d="M4208 4275 c-3 -3 -32 -10 -64 -15 -74 -12 -286 -79 -426 -136 -59
-24 -114 -44 -121 -44 -8 0 -20 -7 -27 -15 -7 -8 -23 -15 -35 -15 -12 0 -28
-7 -35 -15 -6 -8 -59 -37 -116 -66 -57 -28 -104 -55 -104 -59 0 -4 -12 -12
-27 -19 -16 -7 -63 -35 -105 -62 -42 -27 -81 -49 -87 -49 -5 0 -16 -8 -23 -18
-7 -10 -45 -39 -83 -64 -39 -25 -77 -53 -84 -62 -7 -9 -20 -16 -27 -16 -8 0
-14 -4 -14 -8 0 -5 -21 -24 -47 -43 -27 -18 -64 -49 -83 -67 -19 -19 -51 -46
-70 -60 -19 -15 -49 -41 -66 -59 -17 -18 -33 -33 -37 -33 -3 0 -40 -34 -81
-76 -41 -42 -80 -73 -86 -69 -5 3 -10 -1 -10 -11 0 -17 -50 -70 -62 -66 -5 1
-8 -3 -8 -9 0 -6 -17 -28 -37 -48 -21 -21 -61 -62 -90 -93 -29 -30 -55 -53
-58 -50 -3 3 -5 -1 -5 -8 0 -7 -12 -25 -27 -39 -16 -15 -46 -49 -68 -77 -81
-101 -106 -124 -124 -114 -10 5 -46 13 -82 18 -35 6 -109 18 -164 27 -108 18
-301 15 -405 -5 -30 -6 -66 -11 -80 -11 -13 -1 -53 -14 -89 -30 -35 -16 -74
-29 -87 -29 -13 0 -24 -4 -24 -10 0 -5 -7 -10 -15 -10 -8 0 -15 -4 -15 -9 0
-5 -10 -11 -22 -15 -13 -3 -39 -19 -58 -36 -19 -16 -40 -30 -46 -30 -6 0 -29
-17 -50 -37 -22 -21 -66 -62 -97 -92 -54 -50 -107 -111 -107 -123 0 -3 53 -6
117 -7 126 -1 323 -20 323 -30 0 -3 15 -6 33 -7 50 -1 108 -14 103 -23 -3 -4
7 -8 22 -9 40 -1 92 -13 92 -22 0 -4 27 -18 59 -30 33 -12 84 -37 113 -56 48
-31 51 -35 30 -40 -12 -4 -21 -11 -20 -18 3 -13 -33 -76 -44 -76 -4 0 -8 -6
-8 -14 0 -8 -15 -30 -33 -49 -18 -19 -31 -40 -29 -46 2 -6 -1 -11 -6 -11 -5 0
-12 -10 -16 -22 -3 -13 -18 -37 -34 -53 -30 -31 -56 -74 -50 -80 68 -73 223
-225 229 -225 5 0 6 -8 2 -17 -3 -10 -3 -15 2 -11 4 4 13 4 19 -1 7 -6 28 6
59 35 27 24 53 41 59 37 7 -4 9 0 5 10 -5 13 -1 17 14 17 14 0 19 5 16 14 -6
15 49 63 65 58 5 -1 6 2 3 8 -7 12 47 70 65 70 5 0 10 5 10 10 0 6 11 22 25
35 13 14 29 25 35 25 5 0 10 5 10 10 0 17 47 60 65 60 9 0 15 6 13 13 -3 13
36 57 52 57 5 0 10 6 10 14 0 15 58 62 72 58 5 -1 8 4 6 11 -3 14 45 67 62 67
5 0 10 5 10 11 0 15 36 46 62 55 11 3 18 11 16 17 -5 15 35 54 49 50 6 -2 13
7 17 20 3 13 17 29 31 37 14 7 25 17 25 22 0 14 60 68 75 68 8 0 15 7 15 15 0
17 43 65 58 65 5 0 14 11 22 24 8 16 18 22 28 18 11 -4 14 -1 9 11 -4 11 -1
17 8 17 8 0 15 5 15 11 0 17 48 59 67 59 9 0 13 6 10 14 -6 15 49 63 65 58 5
-1 6 2 3 8 -7 12 47 70 65 70 5 0 10 5 10 11 0 17 117 129 135 129 9 0 15 6
13 13 -3 13 36 57 52 57 6 0 10 6 10 14 0 15 58 62 72 58 5 -1 8 4 6 11 -4 17
44 65 60 59 7 -2 12 5 12 15 0 10 10 25 22 33 27 18 88 82 88 92 0 4 6 5 14 2
8 -3 16 4 19 19 4 13 18 29 32 37 14 7 25 16 25 19 0 14 50 63 67 65 10 2 20
11 22 21 4 21 45 65 61 65 5 0 10 5 10 10 0 14 46 60 60 60 5 0 10 4 10 10 0
15 50 60 67 60 9 0 13 6 10 14 -5 14 51 63 66 58 5 -1 5 3 1 9 -7 12 55 71 67
63 4 -2 10 0 14 6 3 5 1 10 -6 10 -7 0 2 14 19 30 18 17 36 30 42 30 5 0 10 5
10 10 0 17 47 60 65 60 9 0 15 5 13 12 -4 12 37 58 52 58 6 0 10 7 10 15 0 8
18 25 40 37 22 13 39 28 37 33 -3 16 46 62 61 57 7 -2 12 5 12 15 0 10 10 25
22 33 12 8 35 29 52 47 16 18 34 29 39 26 6 -4 8 0 4 10 -4 12 -1 17 12 17 13
0 18 8 19 30 1 25 -3 30 -22 30 -12 0 -25 -2 -28 -5z"/>
<path d="M2900 2665 c-707 -707 -1291 -1285 -1297 -1285 -7 0 -20 -12 -29 -26
-17 -25 -16 -26 25 -68 23 -23 38 -45 35 -49 -4 -4 -1 -7 6 -7 8 0 24 -9 36
-21 l22 -20 130 84 c72 47 154 104 182 126 29 23 57 41 64 41 7 0 22 -19 34
-43 12 -23 30 -55 41 -72 23 -38 95 -251 111 -327 13 -68 17 -121 19 -320 0
-88 5 -148 11 -148 20 0 208 199 255 270 28 41 53 77 56 80 9 7 50 90 54 109
2 9 15 47 30 85 79 208 83 471 12 737 l-23 85 42 42 c24 23 46 42 49 42 8 0
218 190 223 201 2 5 10 5 18 2 10 -4 14 -1 13 8 -2 8 60 76 138 151 77 76 137
138 134 138 -3 0 17 22 46 49 29 27 50 51 47 54 -3 2 13 19 36 37 22 17 40 37
40 44 0 7 13 24 28 37 15 13 64 73 107 133 44 59 95 125 113 144 17 20 32 42
32 49 0 7 5 13 11 13 5 0 8 4 5 9 -3 5 1 14 9 21 7 6 28 35 46 63 18 29 44 70
58 92 31 47 111 192 142 255 92 192 107 228 114 260 2 8 19 54 39 102 37 88
42 108 29 108 -5 0 -586 -578 -1293 -1285z"/>
<path d="M1110 1605 c0 -8 -4 -15 -10 -15 -5 0 -10 -6 -10 -13 0 -8 -20 -39
-45 -71 -24 -31 -47 -66 -50 -77 -5 -16 24 -50 144 -169 83 -82 151 -153 151
-157 0 -4 18 -21 40 -36 22 -16 40 -34 40 -41 0 -7 12 -19 26 -26 23 -13 27
-12 35 1 5 9 47 40 93 70 69 43 85 58 84 76 -2 13 -7 23 -13 23 -5 0 -51 42
-101 93 -50 50 -113 113 -140 138 -27 25 -64 63 -83 85 -19 21 -50 51 -69 67
-19 15 -32 27 -28 27 3 0 -3 9 -14 20 -24 24 -50 26 -50 5z"/>
<path d="M913 1310 c-3 -11 -10 -20 -15 -20 -5 0 -7 -4 -3 -9 3 -6 -1 -13 -8
-18 -7 -4 -34 -42 -60 -84 -26 -42 -59 -89 -72 -105 -14 -16 -25 -35 -25 -41
0 -7 -5 -13 -11 -13 -5 0 -8 -4 -4 -9 3 -5 1 -12 -5 -16 -6 -3 -38 -48 -71
-98 -34 -51 -65 -93 -70 -95 -5 -2 -9 -10 -9 -17 0 -7 -5 -15 -11 -17 -5 -2
-11 -11 -13 -20 -1 -10 -11 -28 -23 -40 -17 -19 -92 -128 -148 -217 -5 -9 -14
-19 -20 -23 -5 -4 -6 -8 -1 -8 5 0 0 -10 -11 -22 -11 -13 -42 -56 -68 -97 -26
-41 -53 -81 -61 -89 -22 -21 -33 -47 -31 -69 1 -22 5 -20 82 40 5 4 82 55 170
112 88 58 165 109 171 114 13 11 203 138 243 162 14 9 32 14 40 11 8 -3 12 0
9 8 -3 9 36 41 106 87 60 40 112 73 114 73 3 0 28 17 56 38 28 21 75 53 104
72 44 27 52 37 48 54 -3 12 -6 24 -6 28 0 4 -6 5 -13 3 -7 -3 -89 72 -187 170
-95 96 -178 175 -183 175 -5 0 -11 -9 -14 -20z"/>
</g>
</svg>
