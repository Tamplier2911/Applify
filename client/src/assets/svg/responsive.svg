<?xml version="1.0"?>
<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 64 64" width="512" height="512"><g id="Computer_tablet_smartphone" data-name="Computer tablet smartphone"><path d="M47,30H57V7H7V22H25a5.006,5.006,0,0,1,5,5v7H42.1A5.009,5.009,0,0,1,47,30ZM29,20H27V18h2Zm4,0H31V18h2Zm4,0H35V18h2Z"/><rect x="30" y="36" width="12" height="6"/><path d="M60,35a2.985,2.985,0,0,0-2.239-2.9l-.014,0-.035-.007A2.642,2.642,0,0,0,57,32H56v1a2,2,0,0,1-2,2H50a2,2,0,0,1-2-2V32H47a3,3,0,0,0-3,3V57a3,3,0,0,0,3,3H57a3,3,0,0,0,3-3ZM49,58H47V56h2Zm4,0H51V56h2Zm4,0H55V56h2Z"/><polygon points="30 54 40.5 54 36.333 44 30 44 30 54"/><path d="M28,27a3,3,0,0,0-3-3H22v1a3,3,0,0,1-3,3H13a3,3,0,0,1-3-3V24H7a2.634,2.634,0,0,0-.729.093A2.994,2.994,0,0,0,4,27V57a3,3,0,0,0,3,3H25a3,3,0,0,0,3-3ZM13,58H11V56h2Zm4,0H15V56h2Zm4,0H19V56h2Z"/><path d="M19,26a1,1,0,0,0,1-1V24H12v1a1,1,0,0,0,1,1Z"/><rect x="50" y="32" width="4.001" height="1"/><path d="M60,5a1,1,0,0,0-1-1H5A1,1,0,0,0,4,5V23c.034-.026.8-.481.889-.524.037-.017.073-.036.111-.052V5H59V30.432a4.944,4.944,0,0,1,1,.562Z"/></g></svg>
