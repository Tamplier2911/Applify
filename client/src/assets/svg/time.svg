<svg id="Time" height="512" viewBox="0 0 64 64" width="512" xmlns="http://www.w3.org/2000/svg"><path d="m30 56c.66 0 1.33-.036 2-.089v-2.011c-.671.057-1.34.1-2 .1a24 24 0 1 1 24-24c0 .66-.039 1.329-.1 2h2.007c.053-.67.089-1.34.089-2a26 26 0 1 0 -25.996 26z"/><path d="m46.445 46.168a1 1 0 0 1 1.11 0l1.445.964v-7.132h-4v7.132z"/><path d="m37.535 34-2.667 4h24.264l-2.667-4z"/><path d="m51 40h9v3h-9z"/><path d="m34 40h9v3h-9z"/><path d="m35 60h24a1 1 0 0 0 1-1v-14h-9v4a1 1 0 0 1 -1.555.832l-2.445-1.632-2.445 1.63a1 1 0 0 1 -1.555-.83v-4h-9v14a1 1 0 0 0 1 1z"/><path d="m29 29h2v2h-2z" transform="matrix(.707 -.707 .707 .707 -12.426 30)"/><path d="m32 39a1.006 1.006 0 0 1 .168-.555l4-6a1 1 0 0 1 .832-.445h14.9c.062-.671.1-1.341.1-2a22 22 0 1 0 -22 22c.659 0 1.329-.043 2-.1zm16-10h2v2h-2zm-4.564-13.85 1.414 1.414-1.415 1.415-1.414-1.414zm-31.436 15.85h-2v-2h2zm17-21h2v2h-2zm-12.436 5.15 1.415 1.415-1.414 1.414-1.415-1.415zm0 29.7-1.414-1.414 1.415-1.415 1.414 1.414zm.436-13.85v-2h9.758l2.242-2.242v-11.758h2v11.758l2.535 2.535a1 1 0 0 1 0 1.414l-2.828 2.828a1 1 0 0 1 -1.414 0l-2.535-2.535zm14 19h-2v-2h2z"/></svg>