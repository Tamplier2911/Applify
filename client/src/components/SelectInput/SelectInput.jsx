// import "./SelectInput.scss";
import React from "react";
import PropTypes from "prop-types";

// JS Rendering CSS
import {
  SelectInputContainer,
  SelectInputLabel,
  SelectInputElement,
  SelectInputOptions,
} from "./SelectInputStyles";

/**
 * @JSXComponent
 * @param {function} onInputChange - on change trigger.
 * @param {JSXComponent} label - svg icon that we want to render within select bar.
 * @param {string} count - represent amount of labels to render, eg 1-5 stars for feedback.
 * @param {array} options - array of objects to render select options {name: "", value: ""}.
 * @param {object} otherProps - defaultValue, required etc..
 */

export const SelectInput = ({
  onInputChange,
  label,
  options,
  count,
  ...otherProps
}) => {
  return (
    <SelectInputContainer data-test="select-input">
      {label && count ? (
        <SelectInputLabel columns={count}>
          {Array.from(new Array(Number(count)), (el, i) => i + 1).map((el, i) =>
            React.cloneElement(label, {
              key: `${el + i}`,
              "data-test": "select-input-label",
            })
          )}
        </SelectInputLabel>
      ) : (
        <SelectInputLabel columns={label.length} data-test="select-input-label">
          {label}
        </SelectInputLabel>
      )}
      <SelectInputElement
        {...otherProps}
        onChange={onInputChange}
        data-test="select-input-select"
      >
        {options.map((option, i) => {
          const { name, value } = option;
          return (
            <SelectInputOptions
              data-test="select-input-select-option"
              value={value}
              key={i}
            >
              {name}
            </SelectInputOptions>
          );
        })}
      </SelectInputElement>
    </SelectInputContainer>
  );
};

SelectInput.propTypes = {
  onInputChange: PropTypes.func,
  label: PropTypes.object,
  options: PropTypes.arrayOf(
    PropTypes.shape({
      name: PropTypes.string,
      value: PropTypes.string,
    })
  ),
  count: PropTypes.string,
};

export default SelectInput;
