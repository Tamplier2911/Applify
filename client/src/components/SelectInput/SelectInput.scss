/*

.select-input {
  display: grid;

  &__label {
    display: grid;
    grid-template-columns: repeat(5, min-content);
    grid-column-gap: 2rem;
    justify-content: center;

    padding: 2rem 0rem;
  }

  &__element {
    cursor: pointer;
    font-family: var(--font-title);
    font-size: 1.6rem;
    padding: 0.5rem 1rem;

    // appearance: none;

    outline: none;
    box-shadow: none;
    transition: border 0.3s;

    &:hover {
      border: 0.1rem solid var(--cl-font);
    }

    &::-moz-focus-inner {
      border: 0;
    }

    &:focus {
      outline: none;
    }

    & option {
      color: var(--cl-font);
    }

    & option:checked {
      color: var(--cl-white);
      background-color: var(--cl-font);
    }
  }

  &__option {
  }
}

*/
