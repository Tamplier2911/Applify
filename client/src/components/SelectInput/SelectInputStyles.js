import styled from "styled-components";

const getColumnsAmount = (props) => {
  const { columns } = props;
  return `grid-template-columns: repeat(${columns}, min-content);`;
};

export const SelectInputContainer = styled.div`
  display: grid;
`;

export const SelectInputLabel = styled.div`
  display: grid;
  ${getColumnsAmount}
  grid-column-gap: 2rem;
  justify-content: center;

  padding: 2rem 0rem;
`;

export const SelectInputElement = styled.select`
  cursor: pointer;
  font-family: var(--font-title);
  font-size: 1.6rem;
  padding: 0.5rem 1rem;

  outline: none;
  box-shadow: none;
  transition: border 0.3s;

  &:hover {
    border: 0.1rem solid var(--cl-font);
  }

  &::-moz-focus-inner {
    border: 0;
  }

  &:focus {
    outline: none;
  }

  & option {
    color: var(--cl-select);
  }

  & option:checked {
    color: var(--cl-white);
    background-color: var(--cl-font);
  }
`;

export const SelectInputOptions = styled.option``;
