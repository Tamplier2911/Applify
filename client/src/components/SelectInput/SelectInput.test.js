import React from "react";
import toJson from "enzyme-to-json";

// svg
import { ReactComponent as TestSVG } from "../../assets/svg/admin.svg";

// component
import SelectInput from "./SelectInput";

// utils
import { setShallow, findByTestAttr, checkProps } from "../../tests/testUtils";

describe("<SelectInput />", () => {
  let wrapper;
  const customProps = {
    onInputChange: () => {},
    label: <TestSVG />,
    options: [{ name: "opt1", value: "val1" }],
    count: "1",
  };
  beforeEach(() => {
    wrapper = setShallow(<SelectInput {...customProps} />);
  });

  it("renders without an error", () => {
    const selectInput = findByTestAttr(wrapper, "select-input");
    expect(selectInput).toHaveLength(1);
  });

  it("renders children - select input select", () => {
    const selectInputSelect = findByTestAttr(wrapper, "select-input");
    expect(selectInputSelect).toHaveLength(1);
  });

  it("renders as much labels, as count props equals", () => {
    const customProps = {
      onInputChange: () => {},
      label: <TestSVG />,
      options: [{ name: "opt1", value: "val1" }],
      count: "3",
    };
    wrapper = setShallow(<SelectInput {...customProps} />);
    const selectInputLabel = findByTestAttr(wrapper, "select-input-label");
    expect(selectInputLabel.length).toBe(Number(customProps.count));
  });

  it("renders as much select options, as options length is", () => {
    const customProps = {
      onInputChange: () => {},
      label: <TestSVG />,
      options: [
        { name: "opt1", value: "val1" },
        { name: "opt2", value: "val2" },
      ],
      count: "3",
    };
    wrapper = setShallow(<SelectInput {...customProps} />);
    const selectInputSelectOption = findByTestAttr(
      wrapper,
      "select-input-select-option"
    );
    expect(selectInputSelectOption.length).toBe(customProps.options.length);
  });

  it("recieves expected props", () => {
    const expectedProps = {
      onInputChange: () => {},
      label: <TestSVG />,
      options: [],
      count: "1",
    };
    const err = checkProps(SelectInput, expectedProps, "props");
    expect(err).toBeUndefined();
  });

  it("matches snapshot", () => {
    const selectInput = findByTestAttr(wrapper, "select-input");
    expect(toJson(selectInput)).toMatchSnapshot();
  });
});
