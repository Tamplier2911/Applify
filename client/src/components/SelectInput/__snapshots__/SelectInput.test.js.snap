// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SelectInput /> matches snapshot 1`] = `
<styled.div
  data-test="select-input"
>
  <styled.div
    columns="1"
  >
    <ForwardRef(SvgAdmin)
      data-test="select-input-label"
      key="1"
    />
  </styled.div>
  <styled.select
    data-test="select-input-select"
    onChange={[Function]}
  >
    <styled.option
      data-test="select-input-select-option"
      key="0"
      value="val1"
    >
      opt1
    </styled.option>
  </styled.select>
</styled.div>
`;
